src/core/occupancy.ts

```diff
@@ -110,15 +110,23 @@ export class DefaultOccupancy
    * @param chatApi An instance of the ChatApi.
    * @param logger An instance of the Logger.
    */
-  constructor(roomId: string, realtime: Ably.Realtime, chatApi: ChatApi, logger: Logger) {
+  constructor(roomId: string, realtime: Ably.Realtime, chatApi: ChatApi, logger: Logger, initAfter: Promise<void>) {
     super();
     this._roomId = roomId;
-    this._channel = getChannel(messagesChannelName(roomId), realtime, { params: { occupancy: 'metrics' } });
-    addListenerToChannelWithoutAttach({
-      listener: this._internalOccupancyListener.bind(this),
-      events: ['[meta]occupancy'],
-      channel: this._channel,
+
+    this._channel = initAfter.then(() => {
+      const channel = getChannel(messagesChannelName(roomId), realtime, { params: { occupancy: 'metrics' } });
+      addListenerToChannelWithoutAttach({
+        listener: this._internalOccupancyListener.bind(this),
+        events: ['[meta]occupancy'],
+        channel: channel,
+      });
+      return channel;
     });
+
+    // catch this so it won't send unhandledrejection global event
+    this._channel.catch(() => void 0);
```

Comment:
What events lead to the error being thrown and not caught by using `initAfter.then().catch()` ?

Reply:
if we do
```
this._channel = initAfter.then(...).catch(...)
```
then this._channel will be a promise that always _resolves_ (in case of catch it'll resolve to whatever that returns, likely `undefined`), and is never rejected (assuming the catch block doesn't return `Promise.reject(...)`). We don't want that. We want it to be rejected if release happened before init.

So to _propagate_ the rejection we just don't catch it. But that means the global unhandled promise thing gets triggered if this._channel didn't have a `catch` registered anywhere else in code at the time of rejection.